modules/nlu/src/views/full/intents/UtterancesEditor.tsx

```diff
@@ -138,11 +139,29 @@ export class UtterancesEditor extends React.Component<Props> {
           show={this.state.showSlotMenu}
           onSlotClicked={this.tag.bind(this, editor)}
         />
-        <div className={style.utterances}>{children}</div>
+        <div className={style.utterances} onCopy={this.onCopy}>{children}</div>
       </div>
     )
   }
 
+  onCopy = event => {
+    const selection = document.getSelection().toString()
+    let lines: string[]
+    if (!selection.length) {
+      // Selected the whole component, we put all utterances in the clipboard
+      lines = valueToUtterances(this.state.value)
+    } else {
+      // Partial selection, we remove the heading numbers and empty lines
+      lines = selection.split('\n').map(txt =>
+        txt.replace(/^\d{1,4}$/, '')
+      ).filter(x => x.length)
+    }
+
+    event.clipboardData.setData('text/plain', lines.join('\n'))
+    event.clipboardData.setData('text/html', '<ul>' + lines.map(x => `<li>${x}</li>`).join('') + '</ul>')
```

Comment:
seriously not a fan of copying as a list, is that necessary for pasting in the intent editor ?

Reply:
no it's not necessary, it's when you paste in HTML containers.  let's merge and if you hate it then PR it 👍